src/main/java/com/car/app/controller/CarDao.java

```diff
@@ -39,10 +46,23 @@
 	 * getting all cars
 	 * @return List of cars   list of cars
 	 */
-	public List<Car> findAllCars() {
-		List<Car> resultList = entityManager.createQuery("select c from Car c").getResultList();
+	public PaginationObject<CarDto> findAllCars(int size, int page, String sort, String order, String search, String searchBy) {
+		sort = sort != null && !sort.trim().isEmpty() ? sort : "asc";
```

Comment:
Dealing with strings in dynamic queries is a dangerous approach, please check Criteria API https://www.objectdb.com/java/jpa/query/criteria .